webpack/assets/javascripts/react_app/redux/middlewares/IntervalMiddleware/IntervalMiddleware.js

```diff
@@ -0,0 +1,36 @@
+import { START_INTERVAL, STOP_INTERVAL } from './IntervalConstants';
+import { selectIntervalID } from './IntervalSelectors';
+import { amendActionsPayload } from '../common/helpers';
+import {
+  registeredIntervalException,
+  unregisteredIntervalException,
+} from './IntervalHelpers';
+
+export const IntervalMiddleware = store => next => action => {
+  const { type, payload: { key, method, interval, args = [] } = {} } = action;
+  const state = store.getState();
+
+  if (type === START_INTERVAL) {
```

Comment:
I think I prefer `case` statements instead of having multiple `if`s -- especially if there's no `else` or `else if`.  thoughts?

Reply:
I also tried to use the `switch` statement though eslint doesn't recommend to do declare `const` inside them and `throw` stuff so I decided to use `if`s. another reason is that there are nested `if`s and the code could get more complex.

Reply:
I think the nested `if`s is an argument against `if`, not for. :yum: 

You can prevent the eslint complaining by making a block for each case where you want to declare a const:
```
case START_INTERVAL: {
  const myConst = 'value';
  ...
}

Reply:
I don't mind changing it.. can't argue about style :stuck_out_tongue_closed_eyes:

Reply:
switched